### PR TITLE
    EES-2861 - add all_methodologies_snapshot test

### PR DESCRIPTION
**Pull request by luke-h1**

### CODE REVIEW

File: tests/robot-tests/scripts/create_snapshots.py [hunk 1]
```diff
@@ -118,6 +117,45 @@ def create_data_catalogue_snapshot(public_url) -> str:
...
+                'topic_heading': topic_html.select_one('[id^="details-heading-"]').string,
+                'methodologies': []
+            }
+
+            methodologies = topic_html.select('[id^="methodology-heading-"]') or []
+            for methodology_html in methodologies:
+                methodologies = []
+                methodologies.append(methodology_html.string)
+                
+            topic['methodologies'].append(methodologies)
```
Comment (markhiveit) [inline tests/robot-tests/scripts/create_snapshots.py:150]: So looking at this again, I think L147-150 can be removed and instead you can just have
```
            for methodology_html in methodologies:
                topic['methodologies'].append(methodology_html.string)
```

File: tests/robot-tests/scripts/create_snapshots.py [hunk 2]
```diff
@@ -118,6 +117,45 @@ def create_data_catalogue_snapshot(public_url) -> str:
...
+        topics = theme_html.select('details') or []
+
+        for topic_html in topics:
+            topic = {
+                'topic_heading': topic_html.select_one('[id^="details-heading-"]').string,
+                'methodologies': []
+            }
+
+            methodologies = topic_html.select('[id^="methodology-heading-"]') or []
+            for methodology_html in methodologies:
```
Comment (markhiveit) [inline tests/robot-tests/scripts/create_snapshots.py:146]: Could rename `methodology_html` to `methodology_heading`, just to be a little clearer.

File: tests/robot-tests/scripts/create_snapshots.py [hunk 3]
```diff
@@ -118,6 +117,45 @@ def create_data_catalogue_snapshot(public_url) -> str:
 
     return json.dumps(result, sort_keys=True, indent=2)
 
+def create_all_methodologies_snapshot(public_url) -> str:
+    all_methodologies_url = f"{public_url.rstrip('/')}/methodology"
+    parsed_html = _gets_parsed_html_from_page(all_methodologies_url)
+
+    methodologies_accordion = parsed_html.find(id="publications")
```
Comment (markhiveit) [inline tests/robot-tests/scripts/create_snapshots.py:124]: I wonder of the selector `id="publications"` can be renamed on the frontend to make it more clearer - the other snapshots use `id="themes"`. I assume this is hardcoded in the methodology page frontend code.

**Review by markhiveit — COMMENTED** [review]
You also need to add need to add the new snapshot to `tests/general_public/check_snapshots.robot` and `tests/libs/snapshots.py`.

I would also add the file `tests/snapshots/all_methodologies_snapshot.json`, containing just `[]` -- this might not be necessary, but I don't know how the robot tests would handle not finding the file, so I would add it to be safe.

File: src/explore-education-statistics-frontend/src/modules/methodologies/MethodologyIndexPage.tsx [hunk 4]
```diff
@@ -71,7 +71,11 @@ const MethodologyIndexPage: NextPage<Props> = ({ themes = [] }) => {
               >
                 {topics.map(
                   ({ id: topicId, title: topicTitle, publications }) => (
-                    <Details key={topicId} id={topicId} summary={topicTitle}>
+                    <Details
+                      key={topicId}
+                      id={`methodology-${topicId}`}
```
Comment (markhiveit) [inline src/explore-education-statistics-frontend/src/modules/methodologies/MethodologyIndexPage.tsx:76]: I think this id tag would be better as `topic-${topicId}`?

File: tests/robot-tests/scripts/create_snapshots.py [hunk 5]
```diff
@@ -4,7 +4,6 @@
 import requests
 from bs4 import BeautifulSoup
 
-
```
Comment (markhiveit) [inline tests/robot-tests/scripts/create_snapshots.py:7]: Not sure we follow PEP8 everywhere, but it suggests having two blank lines before and after top-level functions: https://www.python.org/dev/peps/pep-0008/#blank-lines

File: tests/robot-tests/scripts/create_snapshots.py [hunk 6]
```diff
@@ -118,6 +117,45 @@ def create_data_catalogue_snapshot(public_url) -> str:
 
     return json.dumps(result, sort_keys=True, indent=2)
 
+def create_all_methodologies_snapshot(public_url) -> str:
```
Comment (markhiveit) [inline tests/robot-tests/scripts/create_snapshots.py:120]: Another blank line above this function so it adheres to PEP8.

File: tests/robot-tests/scripts/create_snapshots.py [hunk 7]
```diff
@@ -118,6 +117,45 @@ def create_data_catalogue_snapshot(public_url) -> str:
...
+                
+            topic['methodologies'].append(methodologies)
+            
+            theme['topics'].append(topic)
+        
+        result.append(theme)
+    
+    return json.dumps(result, sort_keys=True, indent=2)
+
+
```
Comment (markhiveit) [inline tests/robot-tests/scripts/create_snapshots.py:159]: Only two blank lines are necessary here

File: tests/robot-tests/scripts/create_snapshots.py (hunk 1 above)
Comment (markhiveit) [reply tests/robot-tests/scripts/create_snapshots.py:150]: Incidentally, if you're feeling really fancy, you could use list comprehension and get rid of this for loop completely:
https://www.w3schools.com/python/python_lists_comprehension.asp

File: tests/robot-tests/scripts/create_snapshots.py (hunk 3 above)
Comment (luke-h1) [reply tests/robot-tests/scripts/create_snapshots.py:124]: We can change it if needed. The Accordion with this ID is in `MethodologyIndexPage.tsx` 

```
        <Accordion
          id="publications"
          onSectionOpen={accordionSection => {
            logEvent({
              category: 'Methodologies',
              action: 'Publications accordion opened',
              label: accordionSection.title,
            });
          }}
        >
```

File: src/explore-education-statistics-frontend/src/modules/methodologies/MethodologyIndexPage.tsx (hunk 4 above)
Comment (luke-h1) [reply src/explore-education-statistics-frontend/src/modules/methodologies/MethodologyIndexPage.tsx:76]: Don't think this was even being used. I've removed this ID

File: tests/robot-tests/scripts/create_snapshots.py (hunk 3 above)
Comment (luke-h1) [reply tests/robot-tests/scripts/create_snapshots.py]: I've done this

**Pull request merged**